### DISCUSSION TITLE
[PMode.Leg[1].AddressProtocol 'http' is unsupported]

### DISCUSSION THREAD

**Discussion by AlexandruCiobanu** [phax/phase4#133] [Q&A] [answered]
Hi,
I seem to have a problem after launching peppol webapp in prod mode. If I try sending a message to it I get the following error in logs:
 Following errors are present: [PMode.Leg[1].AddressProtocol 'http' is unsupported] -- com.helger.phase4.servlet.AS4UnifiedResponse.setResponseError(AS4UnifiedResponse.java:59)
Phase4 peppol webapp is running in a tomcat and I use nginx as a proxy and SSL.  Is there any was to suppress this error? all https is managed by teh nginx proxy

Thank you,
Alex

**Comment by phax** [comment]
Hi Alex,
well the Peppol network mandates the usage of `https` for exchanging messages securely. That's why this check is contained, so that you are not accidentally sending to a non-https address.
If you received this `http`-address from an SMP it is a compliance issue.
Can you please let me know why you want to send to an `http` instead of an `https` address?
Thanks, Philip

**Comment by AlexandruCiobanu** [comment]
I think I did not explain properly. I have the peppol webapp setup behind an nginx proxy that handles the https part and registered in peppol test network. I use the peppol client to send payloads to that machine (and I see that the metadata gets resolved correctly with https  and reaches the intended machine) but I see in the webapp log the above message. The webapp internally exposes only the http port but that is why I have teh nginx proxy that handles the https part.

**Reply by phax** [reply, marked as answer]
Hi Alex,
I was scanning through the code and the only reason for this kind of error messages is that your configuration property `phase4.endpoint.address` has a value that starts with `http:` - if you change this one to `https` the problem should be gone. phase4 itself does not know "how" it is operated and what is exposed, so if it uses http or https or whatever. So if you "outsourced" the TLS termination to nginx this is totally fine :)
hth

**Reply by AlexandruCiobanu** [reply]
Great catch! I was not setting that property in application.properties. I was using the old server.address property.
Thank you for the assitance!
